python-jsonschema: Update to 4.15.0

jmarcet opened the pull request:
Maintainer: me
Compile tested: master x86_64
Run tested: master x86_64

Description:
New upstream [release](https://github.com/python-jsonschema/jsonschema/releases).

BKPepe commented:
I downloaded artifact [aarch64_cortex-a53-PR19267-e48dc95f5-packages.zip](https://pipelines.actions.githubusercontent.com/serviceHosts/053f8eac-d998-4f2e-b73b-f310b26e3ddb/_apis/pipelines/1/runs/16257/signedartifactscontent?artifactName=aarch64_cortex-a53-PR19267-e48dc95f5-packages&urlExpires=2022-08-27T14%3A46%3A15.1770064Z&urlSigningMethod=HMACV2&urlSignature=ZHlMDRNOz9Uoi39laESqs8Ii%2Fh%2F16JQQuDRGI2VnB9g%3D)` from the Test Build, which we have here and then by using midnight commander, I take a look at inside ``/usr/lib/python3.10/site-packages``  and it still says unknown for the package name, thus this issue https://github.com/openwrt/packages/issues/19101 is still not solved and I wonder, how did you run test your pull request.

jmarcet commented:
@BKPepe Ditto for #19101, I'll double check what's happening.

jmarcet commented:
@BKPepe I found out what happened.

It is true that there are several python packages I maintain but no longer actively use within OpenWRT, 12 in total. They all were dependencies of `docker-compose` when it was written in Python. Anyhow, I had been building them as modules, installing them and checking I could at least import the packages. That usually caught any change in dependencies or brokenness in the package.

Yet something else happened with jsonschema. I do have a `/usr/local/` bind mount where I have some alpine packages I installed manually, and it turns out I had a few pip modules in there too, `jsonschema-4.4.0` being one of them. So I had been testing the same old version again and again without noticing it.

I've cleaned up everything since there was not a single good reason to have any python modules in there and checked out carefully what had happened with the packaging. In `jsonschema-4.6.0` `setup.cfg` was dropped, the same way `setup.py` had been dropped in version `4.0.0`

Thus in this updated PR I'm adding `setup.cfg` from version `4.5.1` with the email and documentation URL updated. I'm not sure this is a path we want to follow, but I understand it's the only way to have it built with the current python system within OpenWRT.

The package has been broken since version `4.6.0` which went into the tree three months ago. Hence I also wonder if anyone else is using it. I rely on alpine containers for any python code I want to run, since they're easier to maintain than all the individual packages in the OpenWRT tree.

From my side I've reworked how I test these python packages so I can easily spot errors like this one and I'll see to extend those tests.

I can confirm it is now packaged correctly, the standalone `jsonschema` works, and basic functionality from the package also works.

There is something else I don't think suits the OpenWRT way of packaging things. Both `benchmarks` and `tests` are included. Shall I add another patch removing them before installation or avoid installing them through some other means?

```
❯ opkg files python3-jsonschema
Package python3-jsonschema (4.15.0-1) is installed on root and has the following files:
/usr/bin/jsonschema
/usr/lib/python3.10/site-packages/jsonschema-4.15.0-py3.10.egg-info/PKG-INFO
/usr/lib/python3.10/site-packages/jsonschema-4.15.0-py3.10.egg-info/SOURCES.txt
/usr/lib/python3.10/site-packages/jsonschema-4.15.0-py3.10.egg-info/dependency_links.txt
/usr/lib/python3.10/site-packages/jsonschema-4.15.0-py3.10.egg-info/entry_points.txt
/usr/lib/python3.10/site-packages/jsonschema-4.15.0-py3.10.egg-info/requires.txt
/usr/lib/python3.10/site-packages/jsonschema-4.15.0-py3.10.egg-info/top_level.txt
/usr/lib/python3.10/site-packages/jsonschema/__init__.pyc
/usr/lib/python3.10/site-packages/jsonschema/__main__.pyc
/usr/lib/python3.10/site-packages/jsonschema/_format.pyc
/usr/lib/python3.10/site-packages/jsonschema/_legacy_validators.pyc
/usr/lib/python3.10/site-packages/jsonschema/_types.pyc
/usr/lib/python3.10/site-packages/jsonschema/_utils.pyc
/usr/lib/python3.10/site-packages/jsonschema/_validators.pyc
/usr/lib/python3.10/site-packages/jsonschema/benchmarks/__init__.pyc
/usr/lib/python3.10/site-packages/jsonschema/benchmarks/issue232.pyc
/usr/lib/python3.10/site-packages/jsonschema/benchmarks/json_schema_test_suite.pyc
/usr/lib/python3.10/site-packages/jsonschema/cli.pyc
/usr/lib/python3.10/site-packages/jsonschema/exceptions.pyc
/usr/lib/python3.10/site-packages/jsonschema/protocols.pyc
/usr/lib/python3.10/site-packages/jsonschema/schemas/draft2019-09.json
/usr/lib/python3.10/site-packages/jsonschema/schemas/draft2020-12.json
/usr/lib/python3.10/site-packages/jsonschema/schemas/draft3.json
/usr/lib/python3.10/site-packages/jsonschema/schemas/draft4.json
/usr/lib/python3.10/site-packages/jsonschema/schemas/draft6.json
/usr/lib/python3.10/site-packages/jsonschema/schemas/draft7.json
/usr/lib/python3.10/site-packages/jsonschema/schemas/vocabularies.json
/usr/lib/python3.10/site-packages/jsonschema/tests/__init__.pyc
/usr/lib/python3.10/site-packages/jsonschema/tests/_helpers.pyc
/usr/lib/python3.10/site-packages/jsonschema/tests/_suite.pyc
/usr/lib/python3.10/site-packages/jsonschema/tests/fuzz_validate.pyc
/usr/lib/python3.10/site-packages/jsonschema/tests/test_cli.pyc
/usr/lib/python3.10/site-packages/jsonschema/tests/test_deprecations.pyc
/usr/lib/python3.10/site-packages/jsonschema/tests/test_exceptions.pyc
/usr/lib/python3.10/site-packages/jsonschema/tests/test_format.pyc
/usr/lib/python3.10/site-packages/jsonschema/tests/test_jsonschema_test_suite.pyc
/usr/lib/python3.10/site-packages/jsonschema/tests/test_types.pyc
/usr/lib/python3.10/site-packages/jsonschema/tests/test_utils.pyc
/usr/lib/python3.10/site-packages/jsonschema/tests/test_validators.pyc
/usr/lib/python3.10/site-packages/jsonschema/validators.pyc
```
